app/src/main/java/at/roteskreuz/stopcorona/model/repositories/InfectionMessengerRepository.kt

```diff
@@ -172,22 +211,53 @@ class InfectionMessengerRepositoryImpl(
             WarningType.GREEN -> {
                 //we are above risc for the last days!!!
                 if (summary.summationRiskScore >= configuration.dailyRiskThreshold) {
-                    //we must now identify day by day if we are YELLOW or RED
-                    //process all 7 days!!! find both and yellow red quarantine day
-                    //val listOfDaysWithDownloadedFilesSortedByServer = apiInteractor.fetchDailyBatchDiagnosisKeys()
-                    //process the batches and do one of these:
-                    //TODO find time of contact
-                    //this is a fake calculation:
-                    val dayOfExposure = ZonedDateTime.now().minusDays(summary.daysSinceLastExposure.toLong())
-                    quarantineRepository.receivedWarning(WarningType.YELLOW, timeOfContact = dayOfExposure)
-                    var quarantineStatus = quarantineRepository.getQuarantineStatus()
-                } else {
+
+                    //1. let´s remove batches we definitly don´t need
+                    val relevantDailyBatchesParts = fullSession.dailyBatchesParts.filter {
+                        // [   ][   ][ 2 ][   ][now]
+                        // xxxxxxxxxxxxxxx|<- this is the reference date for minusDays(2)
+                        // we want all the xxxxxxxxxxxxxxx
+                        val referenceDate = ZonedDateTime.now().minusDays(summary.daysSinceLastExposure.toLong()).endOfTheDay()
+                        it.intervalStart < referenceDate.toEpochSecond()
+
+                        //org.threeten.bp.ZonedDateTime.ofInstant(org.threeten.bp.Instant.ofEpochSecond(fullSession.dailyBatchesParts.get(0).intervalStart), ZoneId.systemDefault())
+                    }
+                    Timber.d("filtered the relevantDailyBatchesParts to length ${relevantDailyBatchesParts.size} ")
+                    processAndDropNextDayPersistState(relevantDailyBatchesParts, fullSession)
+                }
+                else {
                     Timber.d("We are still WarningType.REVOKE")
```

Comment:
else-branch can be removed

Reply:
Let's keep the logging for error analysis. It will be dropped in production anyway.